events/recorder_histogram.go

```diff
@@ -0,0 +1,71 @@
+package events
+
+import (
+	"time"
+
+	"github.com/mongodb/ftdc"
+	"github.com/mongodb/grip"
+)
+
+type histogramStream struct {
+	point     *PerformanceHDR
+	started   time.Time
+	collector ftdc.Collector
+	catcher   grip.Catcher
+}
+
+// NewHistogramRecorder collects data and stores them with a histogram
+// format. Like the Collapsed recorder, the system saves each data
+// point after a call to Begin.
+//
+// The timer histgrams have a minimum value of 1 microsecond, and a
+// maximum value of 20 minutes, with 5 significant digits. The counter
+// histograms store between 0 and 1 million, with 5 significant
+// digits. The gauges are not stored as integers.
+//
+// The histogram reporter is not safe for concurrent use without a
+// synchronixed wrapper.
+func NewHistogramRecorder(collector ftdc.Collector) Recorder {
+	return &histogramStream{
+		point:     NewHistogramMillisecond(PerformanceGauges{}),
+		collector: collector,
+		catcher:   grip.NewExtendedCatcher(),
+	}
+}
+
+func (r *histogramStream) SetState(val int)   { r.point.Gauges.State = int64(val) }
```

Comment:
This casting is probably unnecessary. Why not make the interface support int64 instead of int?